lib/commands/publish.js

```diff
@@ -196,6 +203,26 @@ class Publish extends BaseCommand {
     }
   }
 
+  async #latestPublishedVersion (spec) {
+    try {
+      const uri = '/' + spec.escapedName
+      const packument = await npmFetch.json(uri, {
+        ...this.npm.flatOptions,
+        spec,
+      })
+      if (typeof packument?.versions === 'undefined') {
+        return null
+      }
+      const ordered = Object.keys(packument?.versions)
+        .map(v => new semver.SemVer(v))
+        .filter(v => !v.prerelease.length)
```

Comment:
```suggestion
        .flatMap(v => {
          const s = new semver.SemVer(v)
          return s.prerelease.length > 0 ? [] : s;
        })
```
if you want to do it in a single pass; this might matter for performance of packages with lots of versions published, like react, TS, etc